runtime/GNUmakefile

```diff
@@ -28,6 +28,8 @@
 .NOTPARALLEL all: phase_omr
 clean: clean_phase_omr
 
+phase_omr: generate_jvmti
```

Comment:
OMR should not use `jvmti.h` so this is inappropriate.
Instead add a dependency on `generate_jvmti` in a suitable `module.xml` - perhaps `runtime/vm/module.xml`?

Reply:
Although omr doesn't use `jvmti.h`, it needs to include `oti/j9nonbuilder.h` when it compiles ObjectAllocationInterface.cpp. `j9nonbuilder.h` needs to include `jvmti.h`.

Reply:
Can you share the include chain that goes from `ObjectAllocationInterface.cpp` to `jvmti.h`, please?

Reply:
I have sent you through slack.

Reply:
For those following along, you suggested that having `j9include` depend on `generate_jvmti` might work: I like that answer much better.

Reply:
After I tried it out several times, add an dependency between `j9include` and `generate_jvmti` can't solve the problem. We still need to run generated `makefile` in `include` to get the generated file.
          Besides, `buildtools.mk` is the last file we can modify before entering `omr` part. I add a new target `includegen` before entering `omr`, here is what I think:
          When we runs makefiles in `omr`, `omr` will use command `-I ../include`, which means it search source files `runtime/include`, so generation required files in `runtime/include` just before this step in `buildtools.mk` might be reasonable here.
          We can still use target `includegen` later if we need to generate other files.